tests/load/bigquery/test_bigquery_streaming_insert.py

```diff
@@ -0,0 +1,33 @@
+import dlt
+from dlt.destinations.impl.bigquery.bigquery_adapter import bigquery_adapter
+from tests.pipeline.utils import assert_load_info
+
+
+def test_bigquery_streaming_insert():
```

Comment:
please write a test where you use merge write disposition and use nested data. we need to check if merge sql jobs run

Reply:
Seems it's not so easy with merge disposition. Streaming means buffers are used in BigQuery. Merge is trying to update data, to rearrange, but it's not allowed for 90 mins after streaming insert. Only reading.

Reply:
@rudolfix, so, yes, merge is not that easy with streaming. It may take from several minutes to 1.5 h for the inserted data to become fully available. Thus, merge will work, but its speed is limited (because it does `TRUNCATE`). Streaming insert is just not intended for that kind of operations.

A warning in the docs will be needed, I guess. And something smart for the tests as well... Thinking about it...

Reply:
@IlyaFaer this will be super confusing for our users! yeah I knew that stream insert is doing internal batching but if timeouts are that long it is really not suitable for what we do. my proposition is to ditch this and replace with https://github.com/z3z1ma/target-bigquery/blob/main/target_bigquery/storage_write.py which let's you flush the cache.

for append-only streaming insert we just did this: https://dlthub.com/docs/examples/custom_destination_bigquery/